src/listen/ldap_sync/proto_ldap_sync_ldap.c

```diff
@@ -0,0 +1,1325 @@
+/*
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, write to the Free Software
+ *   Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/**
+ * $Id$
+ * @file proto_ldap_sync_ldap.c
+ * @brief LDAP sync handler.
+ *
+ * @copyright 2022 Network RADIUS SARL (legal@networkradius.com)
+ */
+#define LOG_PREFIX "proto_ldap_sync_ldap"
+
+#include <freeradius-devel/server/protocol.h>
+#include <freeradius-devel/server/request.h>
+#include <freeradius-devel/io/listen.h>
+#include <freeradius-devel/io/application.h>
+#include <freeradius-devel/unlang/call.h>
+#include <freeradius-devel/util/dbuff.h>
+
+#include "proto_ldap_sync_ldap.h"
+#include "rfc4533.h"
+#include "persistent_search.h"
+#include "active_directory.h"
+
+extern fr_app_io_t proto_ldap_sync_ldap;
+extern fr_app_io_t proto_ldap_sync_child;
+
+static CONF_PARSER sasl_mech_static[] = {
```

Comment:
The majority of this config stuff is identical to the configuration for rlm_ldap, could you move it into a common library and then have it work on offsets of `fr_ldap_config_t` then strip it out of rlm_ldap as well.

Reply:
This one can just operate on `fr_ldap_sasl_t`